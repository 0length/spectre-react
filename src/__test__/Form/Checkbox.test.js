import React from 'react'
import { render } from 'enzyme'

import { Checkbox } from '../..'

describe('Checkbox', () => {
  it('Should match the default snapshot', () => {
    const wrapper = render(<Checkbox />)

    expect(wrapper).toMatchSnapshot()
  })

  it('Should match the snapshot with a label', () => {
    const wrapper = render(<Checkbox label='A Label' />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Should not render children', () => {
    const wrapper = render(
      <Checkbox>
        <p>hello world</p>
      </Checkbox>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('Should pass classnames', () => {
    const wrapper = render(<Checkbox className='extra' />)

    expect(wrapper.hasClass('extra')).toBe(true)
    expect(wrapper.hasClass('form-checkbox')).toBe(true)
  })

  it('Should render the sm size', () => {
    const wrapper = render(<Checkbox className='extra' small />)

    expect(wrapper.hasClass('extra')).toBe(true)
    expect(wrapper.hasClass('form-checkbox')).toBe(true)
    expect(wrapper.hasClass('input-sm')).toBe(true)
  })

  it('Should render the lg size', () => {
    const wrapper = render(<Checkbox className='extra' small />)

    expect(wrapper.hasClass('extra')).toBe(true)
    expect(wrapper.hasClass('form-checkbox')).toBe(true)
    expect(wrapper.hasClass('input-sm')).toBe(true)
  })

  it('Should render as a custom element', () => {
    const wrapper = render(<Checkbox renderAs='main' />)

    expect(wrapper.hasClass('form-checkbox')).toBe(true)
    expect(wrapper).toMatchSnapshot()
  })
})
