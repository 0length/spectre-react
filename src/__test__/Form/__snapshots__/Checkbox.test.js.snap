// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Should match the default snapshot 1`] = `
<label
  class="form-checkbox"
>
  <input
    class="form-checkbox"
    type="checkbox"
  />
  <i
    class="form-icon"
  />
</label>
`;

exports[`Checkbox Should match the snapshot with a label 1`] = `
<label
  class="form-checkbox"
>
  <input
    class="form-checkbox"
    type="checkbox"
  />
  <i
    class="form-icon"
  />
  A Label
</label>
`;

exports[`Checkbox Should not render children 1`] = `
<label
  class="form-checkbox"
>
  <input
    class="form-checkbox"
    type="checkbox"
  />
  <i
    class="form-icon"
  />
</label>
`;

exports[`Checkbox Should render as a custom element 1`] = `
<label
  class="form-checkbox"
>
  <main
    class="form-checkbox"
    type="checkbox"
  />
  <i
    class="form-icon"
  />
</label>
`;
